src/Service/CaseManager.php

```diff
@@ -99,4 +103,47 @@ public function newCase(CaseEntity $caseEntity, Board $board): CaseEntity
 
         return $caseEntity;
     }
+
+    public function updateDeadlineBooleans($isDryRun)
+    {
+        // We do this in batches as there might be a large number of cases
+        // @see https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/batch-processing.html#iterating-results
+        $batchSize = 20;
+        $i = 1;
+        $query = $this->caseRepository->createQueryBuilder('c')->getQuery();
+
+        $today = new DateTime('today');
+
+        $this->logger->info('Today: '.$today->format('d/m/Y'));
+
+        foreach ($query->toIterable() as $case) {
+            assert($case instanceof CaseEntity);
+            $hearingDeadline = $case->getFinishHearingDeadline();
+            $processingDeadline = $case->getFinishProcessingDeadline();
+
+            if ($hearingDeadline < $today) {
+                $this->logger->info('Changing case '.$case->getCaseNumber().' with hearing deadline: '.$hearingDeadline->format('d/m/Y'));
+
+                if (!$isDryRun) {
+                    $case->setHasReachedHearingDeadline(true);
+                }
+
+                if ($processingDeadline < $today) {
```

Comment:
Is this at the right indentation level? Because this will only be executed if the hearing deadline is exceeded.

Reply:
When we get the validators working hearing deadlines will always be smaller (come earlier) or equal to the processing deadline. 

In that case the process deadline would never be exceeded if the hearing deadline was not exceeded. So it would be unnecessary to check `$processingDeadline < $today` for cases that does not have an exceeded hearing deadline.